src/3rd_party_config.c

```diff
@@ -0,0 +1,296 @@
+ * Once a section is completely read, it is stored in the global lists.
+ *
+ * @param section A string containing section of the file
+ * @param key A string containing key
+ * @param value A string containing value
+ *
+ * @returns True for every match of section with REPO or GLOBAL, false otherwise
+ */
+bool parse_key_values(char *section, char *key, char *value, void UNUSED_PARAM *data)
+{
+	static int repo_field_count = 0;
+	static int global_field_count = 0;
+	bool ret = false;
+
+	if (!strncmp(section, "REPO", strlen(section))) {
+		if (key && value) {
+			static struct repo *repo_var;
+			if (repo_field_count == 0) {
+				repo_var = (struct repo *)malloc(sizeof(struct repo));
+			}
+			if (!strncmp(key, "repo-name", strlen(key))) {
+				repo_var->repo_name = strdup_or_die(value);
+				repo_field_count++;
+				ret = true;
+			}
+
+			if (!strncmp(key, "repo-url", strlen(key))) {
+				repo_var->url = strdup_or_die(value);
+				repo_field_count++;
+				ret = true;
+			}
+
+			if (repo_field_count == MAX_REPO_ALLOWED_FIELDS) {
+				repos_kv = list_append_data(repos_kv, repo_var);
+				repo_field_count = 0;
+			}
+		}
+	} else if (!strncmp(section, "GLOBAL", strlen(section))) {
+		if (key && value) {
+			static struct global *global_var;
+			if (global_field_count > 2) {
+				return false;
+			}
+
+			if (!strncmp(key, "value1", strlen(key))) {
+				global_var->value1 = strdup_or_die(value);
+				global_field_count++;
+				ret = true;
+			}
+
+			if (!strncmp(key, "value2", strlen(key))) {
+				global_var->value2 = strdup_or_die(value);
+				global_field_count++;
+				ret = true;
+			}
+		}
+	}
+	return ret;
+}
+
+static int write_section(char *name)
+{
+	int ret = 0, fd;
+	ssize_t write_line_len;
+
+	fd = open(repo_config_file_path, O_WRONLY | O_APPEND);
+	if (fd < 0) {
+		return -1;
+	}
+
+	write_line_len = strlen(name);
+
+	if (write_line_len != write(fd, name, write_line_len)) {
+		error("error writing to file %d", errno);
+		ret = -1;
+	}
+
+	close(fd);
+	return ret;
+}
+
+static int write_kv(char *key, char *value)
+{
+	char *write_line;
+	int ret = 0;
+	ssize_t write_line_len;
+
+	int fd = open(repo_config_file_path, O_WRONLY | O_APPEND);
+	if (fd < 0) {
+		return -1;
+	}
+
+	string_or_die(&write_line, "%s=%s\n", key, value);
+	write_line_len = strlen(write_line);
+
+	if (write_line_len != write(fd, write_line, write_line_len)) {
+		error("error writing to file %d", errno);
+		ret = -1;
+	}
+
+	close(fd);
+	free_string(&write_line);
+	return ret;
+}
+
+static int write_empty_line(void)
+{
+	int ret = 0;
+	char *write_line = "\n";
+	ssize_t write_line_length = strlen(write_line);
+
+	int fd = open(repo_config_file_path, O_WRONLY | O_APPEND);
+	if (fd < 0) {
+		return -1;
+	}
+
+	if (write_line_length != write(fd,
+				       write_line,
+				       write_line_length)) {
+		error("error writing to file %d", errno);
+		ret = -1;
+	}
+
+	close(fd);
+	return ret;
+}
```

Comment:
the code for `write_section`, `write_kv` and `write_empty_line` is basically the same, why not just one function that receives the line to be written, like:
`config_append_line(const char *line)`
so we don't duplicate code.

Reply:
ya... makes good sense. I think I didn't push myself for this. I think @otaviobp mentioned something about creating two separate functions(one for key-value, other for section) very early on so I went down this path.  Maybe he had a particular reason. Wait for him to comment

Reply:
Changing this and addressing your concerns in next fix.